timeseries/src/autogluon/timeseries/evaluator.py

```diff
@@ -130,7 +136,7 @@ def _rmse(self, y_true: pd.Series, predictions: TimeSeriesDataFrame) -> float:
     def _mase(self, y_true: pd.Series, predictions: TimeSeriesDataFrame) -> float:
         y_pred = self._get_median_forecast(predictions)
         mae = mae_per_item(y_true=y_true, y_pred=y_pred)
-        return self._safemean(mae / self._past_naive_1_error)
+        return self._safemean(mae / self._past_naive_error)
```

Comment:
If `self._past_naive_error` is 0 or `None`, I guess some error gonna be raised? Can we handle this gracefully here? Also, wondering if there is any case `self._safemean` gonna return `NaN`. If so, would that be expected?

Reply:
- If `self._past_naive_error` equals to 0 for **some** time series, the scaled loss for them will equal `inf` and will be filtered out by `_safemean` (i.e., they won't affect the total loss computation). This happens, e.g., for constant time series. GluonTS will just return `NaN` loss for the entire dataset in such case. I see following options:
  1. Raise an exception like `The provided dataset contains constant time series. The chosen evaluation metric MASE is undefined in this case. Please choose a different eval_metric or remove the constant time series from the dataset`.
  2. Replace 0 -> 1 in `self._past_naive_error` (do not scale the loss for the constant series)
  3. Keep as-is
  
  Do you have any preferences here?
- The case with `self._past_naive_error is None` should never happen, but if it does, an exception will be raised in https://github.com/autogluon/autogluon/pull/2970/files#diff-1711922b22b494da6f1714235d9ac888b754298c639be656bedd0ca35c623007R213
- `self._safemean` can only return `NaN` if all time series have `NaN` loss (e.g., all time series are constant). This seems like a rare/pathological edge case. Do you think we should handle this somehow?

Reply:
See [discussion](https://github.com/awslabs/gluonts/issues/1135) on the same topic from GluonTS. I think we shall follow the same standard and implement MASE which is truthful to its [definition](https://otexts.com/fpp3/accuracy.html)

Reply:
I just double-checked and compared our implementation (this PR) to GluonTS, my previous statement was incorrect. As far as I can tell our code & GluonTS produce identical results:
- If `_past_naive_error` equals to 0 for **some** time series, GluonTS ignores them and reports average MASE for time series where `_past_naive_error > 0`.
- If `_past_naive_error` equals to 0 for **all** time series, GluonTS returns NaN.
Do you think there are any other edge cases that we should check?

Reply:
Fine with it as long as it is a truthful impl of MASE. BTW, we shall fix the `ZeroDivisionError` issue when `self._past_naive_error` is 0 here

Reply:
I just double-checked: if `self._past_naive_error` contains 0 values, pandas doesn't raise any exceptions or warnings and just stores the result as `NaN`. These `NaN`s are then discarded by the `self._safemean` method, so the overall result is identical to the output of GluonTS / compatible with the textbook definition of MASE.

I think `ZeroDivisionError` can only [occur if the underlying `dtype` is `object`](https://stackoverflow.com/questions/72953622/pandas-dataframe-conditional-formatting-still-leads-to-division-by-zero-error/72953731#72953731), but this should never happen as `TimeSeriesDataFrame` can only contain numeric dtypes.